### PR TITLE
Document that `Trivia` cannot represent invalid UTF-8

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SwiftSyntax/Syntax.swift [hunk 1]
```diff
@@ -483,6 +489,12 @@ public extension SyntaxProtocol {
   /// The trailing trivia of this syntax node. Trailing trivia is attached to
   /// the last token syntax contained by this node. Without such token, this
   /// property will return nil.
+  ///
+  /// Note: `Trivia` is not able to represent invalid UTF-8 sequences. To get
+  /// the leading trivia text including all invalid UTF-8 sequences, use
+  /// ```
+  /// node.syntaxTextBytes[(node.byteSize - node.trailingTriviaLength.utf8Length)...]
```
Comment (rintaro) [inline Sources/SwiftSyntax/Syntax.swift:496]: Maybe
```swift
node.syntaxTextBytes.suffix(node.trailingTriviaLength.utf8Length)
```
looks easier. But it doesn't really matter

**Review by rintaro — APPROVED** [review]

File: Sources/SwiftSyntax/Syntax.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/SwiftSyntax/Syntax.swift:496]: Actually, there is no `suffix` on `Array`. But I took the suggestion for `prefix`.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**